outscale/transport.go

```diff
@@ -24,7 +24,8 @@ func (t *transport) sign(req *http.Request, body []byte) error {
 }
 
 func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
-	body, err := ioutil.ReadAll(req.Body)
+	body, err := io.ReadAll(req.Body)
+	defer req.Body.Close()
```

Comment:
Is this code really used ?

Here, you consume the request body, and when you call t.transport.RoundTrip(req), the body sent to the server should be empty.

As a side note, the client do not need to Close the request Body, stdlib already does that.